### PR TITLE
Fix for ficus

### PR DESCRIPTION
**Pull request by flying-pi**

### CODE REVIEW

File: rg_instructor_analytics/views.py [hunk 1]
```diff
@@ -23,10 +23,16 @@
...
+try:
+    from lms.djangoapps.grades.new.course_grade_factory import CourseGradeFactory
+except:
+    from lms.djangoapps.grades.new.course_grade import CourseGradeFactory
+
 from lms.djangoapps.grades.new.course_grade_factory import CourseGradeFactory
 from opaque_keys import InvalidKeyError
 from opaque_keys.edx.keys import CourseKey
-from openedx.core.djangoapps.plugin_api.views import EdxFragmentView
+from web_fragments.views import FragmentView
```
Comment (alexbojko) [inline rg_instructor_analytics/views.py]: ditto

File: rg_instructor_analytics/views.py [hunk 2]
```diff
@@ -23,10 +23,16 @@
 from django.utils.translation import ugettext as _
 from django.views.generic import View
 from edxmako.shortcuts import render_to_string
+
```
Comment (alexbojko) [inline rg_instructor_analytics/views.py]: pls order imports according to these rules https://www.python.org/dev/peps/pep-0008/#imports

File: rg_instructor_analytics/plugins.py [hunk 3]
```diff
@@ -18,10 +19,39 @@ class InstructorAnalyticsDashboardTab(TabFragmentViewMixin, EnrolledTab):
...
+    def fragment_view(self):
+        """
+        Returns the view that will be used to render the fragment.
+        """
+        if not self._fragment_view:
+            self._fragment_view = get_storage_class(self.fragment_view_name)()
+        return self._fragment_view
+
+    def render_to_fragment(self, request, course, **kwargs):
+        """
```
Comment (alexbojko) [inline rg_instructor_analytics/plugins.py:54]: ditto

File: rg_instructor_analytics/plugins.py [hunk 4]
```diff
@@ -18,10 +19,39 @@ class InstructorAnalyticsDashboardTab(TabFragmentViewMixin, EnrolledTab):
...
+
+    @property
+    def url_slug(self):
+        """
+        Returns the slug to be included in this tab's URL.
+        """
+        return "tab/" + self.type
+
+    @property
+    def fragment_view(self):
```
Comment (alexbojko) [inline rg_instructor_analytics/plugins.py]: one line pls

File: rg_instructor_analytics/plugins.py [hunk 5]
```diff
@@ -18,10 +19,39 @@ class InstructorAnalyticsDashboardTab(TabFragmentViewMixin, EnrolledTab):
...
+    def __init__(self, tab_dict):
+        super(InstructorAnalyticsDashboardTab, self).__init__(tab_dict)
+        self._fragment_view = None
+
+    @property
+    def url_slug(self):
+        """
+        Returns the slug to be included in this tab's URL.
+        """
+        return "tab/" + self.type
```
Comment (alexbojko) [inline rg_instructor_analytics/plugins.py]: probably it should be better to use ```"tab/{}".format(self.type)```

File: rg_instructor_analytics/plugins.py [hunk 6]
```diff
@@ -18,10 +19,39 @@ class InstructorAnalyticsDashboardTab(TabFragmentViewMixin, EnrolledTab):
...
         """
         return bool(user and has_access(user, 'staff', course, course.id))
+
+    def __init__(self, tab_dict):
+        super(InstructorAnalyticsDashboardTab, self).__init__(tab_dict)
+        self._fragment_view = None
+
+    @property
+    def url_slug(self):
+        """
```
Comment (alexbojko) [inline rg_instructor_analytics/plugins.py:39]: same as above

File: rg_instructor_analytics/plugins.py [hunk 7]
```diff
@@ -2,12 +2,13 @@
...
+from courseware.tabs import CourseTab
 from django.utils.translation import ugettext_noop
-from xmodule.tabs import TabFragmentViewMixin
+from django.core.files.storage import get_storage_class
 
 
-class InstructorAnalyticsDashboardTab(TabFragmentViewMixin, EnrolledTab):
+
+class InstructorAnalyticsDashboardTab(CourseTab):
     """
```
Comment (alexbojko) [inline rg_instructor_analytics/plugins.py]: one line comment should be fit in one line

File: rg_instructor_analytics/plugins.py [hunk 8]
```diff
@@ -18,10 +18,40 @@ class InstructorAnalyticsDashboardTab(TabFragmentViewMixin, EnrolledTab):
...
     @classmethod
     def is_enabled(cls, course, user=None):
         """
         Return true if the specified user has staff access.
         """
         return bool(user and has_access(user, 'staff', course, course.id))
+
+    def __init__(self, tab_dict):
+        """
+        Counstruct tab.
```
Comment (alexbojko) [inline rg_instructor_analytics/plugins.py:32]: one line pls

File: rg_instructor_analytics/plugins.py [hunk 9]
```diff
@@ -18,10 +18,40 @@ class InstructorAnalyticsDashboardTab(TabFragmentViewMixin, EnrolledTab):
     body_class = "instructor-analytics-tab"
     is_dynamic = True
     fragment_view_name = 'rg_instructor_analytics.views.InstructorAnalyticsFragmentView'
+    view_name = 'instructor_analytics_dashboard'
 
     @classmethod
     def is_enabled(cls, course, user=None):
         """
```
Comment (alexbojko) [inline rg_instructor_analytics/plugins.py:25]: one line please

File: rg_instructor_analytics/plugins.py [hunk 10]
```diff
@@ -18,10 +18,40 @@ class InstructorAnalyticsDashboardTab(TabFragmentViewMixin, EnrolledTab):
...
+        """
+        super(InstructorAnalyticsDashboardTab, self).__init__(tab_dict)
+        self._fragment_view = None
+
+    @property
+    def url_slug(self):
+        """
+        Return the slug to be included in this tab's URL.
+        """
+        return "tab/" + self.type
```
Comment (alexbojko) [inline rg_instructor_analytics/plugins.py]: my proposition to use "tab/{}".format(self.type)

**Review by alexbojko — CHANGES_REQUESTED** [review]

File: rg_instructor_analytics/plugins.py (hunk 9 above)
Comment (flying-pi) [reply rg_instructor_analytics/plugins.py:25]: According to style guide http://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python-guidelines.html, we use multiline style

File: rg_instructor_analytics/plugins.py (hunk 8 above)
Comment (flying-pi) [reply rg_instructor_analytics/plugins.py:32]: ditto

File: rg_instructor_analytics/plugins.py (hunk 6 above)
Comment (flying-pi) [reply rg_instructor_analytics/plugins.py:39]: ditto

File: rg_instructor_analytics/plugins.py (hunk 10 above)
Comment (flying-pi) [reply rg_instructor_analytics/plugins.py]: Thank, I fixed it.

File: rg_instructor_analytics/plugins.py (hunk 4 above)
Comment (flying-pi) [reply rg_instructor_analytics/plugins.py]: ditto

File: rg_instructor_analytics/plugins.py (hunk 3 above)
Comment (flying-pi) [reply rg_instructor_analytics/plugins.py:54]: ditto

File: rg_instructor_analytics/plugins.py (hunk 9 above)
Comment (alexbojko) [reply rg_instructor_analytics/plugins.py:25]: Ok. thank you for information!

File: README.md [hunk 11]
```diff
@@ -1,7 +1,7 @@
 # rg_instructor_analytics
 
 ## Installation
-* Add `rg_instructor_analytics` to the installed app (i.e. in `edx-platform/lms/envs/common.py`)
+* Add `rg_instructor_analytics` and `web_fragments` to the installed app (i.e. in `edx-platform/lms/envs/common.py`)
```
Comment (alexbojko) [inline README.md]: Did you mean INSTALLED_APPS instead of installed app?

File: rg_instructor_analytics/plugins.py (hunk 3 above)
Comment (alexbojko) [reply rg_instructor_analytics/plugins.py:54]: thank you for this fix!

**Review by alexbojko — APPROVED** [review]

**Pull request merged**